frontend/src/scenes/billing/billingLogic.ts

```diff
@@ -55,11 +70,36 @@ export const billingLogic = kea<billingLogicType<PlanInterface, BillingSubscript
                 return color
             },
         ],
-    }),
+        alertToShow: [
+            (s) => [s.eventAllocation, userLogic.selectors.user, s.urlPath],
+            (
+                eventAllocation: FormattedNumber | null | undefined,
+                user: UserType,
+                urlPath: string
+            ): 'setup_billing' | 'usage_near_limit' | undefined => {
+                // Determines which billing alert/warning to show to the user (if any)
+
+                // Priority 1: In-progress incomplete billing setup
+                if (user?.billing?.should_setup_billing && user?.billing.subscription_url) {
+                    return 'setup_billing'
+                }
+
+                // Priority 2: Event allowance near limit
+                if (
+                    urlPath !== '/organization/billing' &&
```

Comment:
I think this would be more resilient if it compared against current scene from `sceneLogic`.